homeassistant/components/mobile_app/webhook.py

```diff
@@ -685,6 +698,15 @@ async def webhook_update_sensor_states(
             continue
 
         sensor[CONF_WEBHOOK_ID] = config_entry.data[CONF_WEBHOOK_ID]
+
+        if unique_store_key in hass.data[DOMAIN][DATA_PENDING_UPDATES]:
+            # Replace existing pending update with the latest sensor data.
+            # This can occur when an entity was recently enabled but not yet initialized.
+            hass.data[DOMAIN][DATA_PENDING_UPDATES][unique_store_key] = sensor
+
+        # Send the dispatcher signal after storing the pending update to handle race conditions.
+        # If the entity initializes before storing the update,
+        # the signal ensures the entity receives the update immediately.
         async_dispatcher_send(
             hass,
             f"{SIGNAL_SENSOR_UPDATE}-{unique_store_key}",
```

Comment:
@MartinHjelmare this is where I make sure that we keep the pending_update up to date with the latest value. I did the test where I updated the volume of the device before the mobile_app is reloaded. It reported properly the latest value I had set on the device not the first once when I registered the sensor.